File: PizzaDelegate/PizzaDelegate/Components/Vegetarian.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PizzaDelegate
{
    class Vegetarian : IVegetarian
    {
        public bool CheckIfChangeable(List<string> ingredientList)
        {
            return (ingredientList.Contains("pepperoni") || ingredientList.Contains("salami")
                || ingredientList.Contains("salmon"));
        }
        public List<string> ReplaceIngredients(List<string> ingredientList)
        {
            ingredientList[ingredientList.FindIndex(ind => ind.Equals("pepperoni") ||
                        ind.Equals("salami") || ind.Equals("salmon"))] = "mushroom";
            return ingredientList;
        }
        public void RemoveIngredients(List<string> ingredientList)
        {
            ingredientList.RemoveAll(t => t == "pepperoni" || t == "salami" || t == "salmon");
        }
    }
}

File: pirma/KotlinDelegate/src/Vegetarian.kt

class Vegetarian() : IVegetarian {
    override fun CheckIfChangeable(ingredientList: MutableList<String>): Boolean{
		var bol : Boolean = true
		for(item in ingredientList)
			{
				if(item == "pepperoni" || item == "salami" || item == "salmon")
					bol = true
				break
			}
		return bol
	}
    override fun ReplaceIngredients(ingredientList: MutableList<String>) : MutableList<String>{
		
		for(item in ingredientList)
			{
				if(item == "pepperoni")
					ingredientList[ingredientList.lastIndexOf("pepperoni")] =  "mushroom"
				if(item == "salami")
					ingredientList[ingredientList.lastIndexOf("salami")] =  "mushroom"
			}
		return ingredientList
	}
    override fun RemoveIngredients(ingredientList: MutableList<String>){
		ingredientList.remove("pepperoni")
		ingredientList.remove("salmon")
		ingredientList.remove("salami")
	}
}